src/tools/perf/perftest_mad.c

```diff
@@ -0,0 +1,695 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "perftest_mad.h"
+#include "perftest.h"
+
+#include <ucs/sys/string.h>
+#include <ucs/sys/sys.h>
+#include <ucs/sys/sock.h>
+#include <ucs/debug/log.h>
+#include <ucs/sys/iovec.inl>
+
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <netdb.h>
+#include <sys/poll.h>
+#include <linux/types.h> /* __be64 */
+
+#include <infiniband/mad.h>
+#include <infiniband/umad.h>
+#include <infiniband/umad_types.h>
+
+#define PERFTEST_RTE_CLASS      (IB_VENDOR_RANGE2_START_CLASS + 0x10)
+#define PERFTEST_RTE_OPENIB_OUI IB_OPENIB_OUI
+
+typedef struct perftest_mad_rte_group {
+    struct ibmad_port *mad_port;
+    ib_portid_t       dst_port;
+    int               is_server;
+} perftest_mad_rte_group_t;
+
+static unsigned mad_magic = 0xdeadbeef;
+
+static ucs_status_t
+perftest_mad_get_remote_port(void *umad, ib_portid_t *remote_port)
+{
+    ib_mad_addr_t *mad_addr = umad_get_mad_addr(umad);
+
+    if (mad_addr == NULL) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+    if (ib_portid_set(remote_port, ntohs(mad_addr->lid), 0, 0)) {
+        return UCS_ERR_NO_DEVICE;
+    }
+    return UCS_OK;
+}
+
+static size_t perftest_mad_iov_size(const struct iovec *iovec, int iovcnt)
+{
+    size_t size = 0;
+    while (iovcnt-- > 0) {
+        size += iovec->iov_len;
+        iovec++;
+    }
+    return size;
+}
+
+static ucs_status_t perftest_mad_sendv(perftest_mad_rte_group_t *mad,
+                                       const struct iovec *iovec, int iovcnt)
+{
+    ucs_status_t status = UCS_OK;
+    ib_rmpp_hdr_t rmpp = {
+        /* Always active, even when data_size <= IB_VENDOR_RANGE2_DATA_SIZE */
+        .flags = IB_RMPP_FLAG_ACTIVE,
+    };
+    ib_rpc_t rpc       = {};
+    size_t data_size   = perftest_mad_iov_size(iovec, iovcnt);
+    size_t size        = umad_size() + IB_VENDOR_RANGE2_DATA_OFFS + data_size;
```

Comment:
minor: can align by `=`

Reply:
aligned `=` at `... rmpp = {` and below with `status = UCS_OK`.